src/rpc/blockchain.cpp

```diff
@@ -477,7 +477,8 @@ static std::vector<RPCResult> MempoolEntryDescription() { return {
         {RPCResult{RPCResult::Type::STR_HEX, "transactionid", "parent transaction id"}}},
     RPCResult{RPCResult::Type::ARR, "spentby", "unconfirmed transactions spending outputs from this transaction",
         {RPCResult{RPCResult::Type::STR_HEX, "transactionid", "child transaction id"}}},
-    RPCResult{RPCResult::Type::BOOL, "bip125-replaceable", "Whether this transaction could be replaced due to BIP125 (replace-by-fee)"},
+    RPCResult{RPCResult::Type::BOOL, "bip125-replaceable", "Whether this transaction could be replaced due to BIP125 (replace-by-fee) (DEPRECATED, use 'replaceable' field instead. Returned only if config option 'deprecatedrpc=bip125' is set)"},
```

Comment:
```suggestion
    RPCResult{RPCResult::Type::BOOL, "bip125-replaceable", "Identical to 'replacable' field. Does *not* return the BIP125 (replace-by-fee) status (DEPRECATED. Returned only if config option 'deprecatedrpc=bip125' is set)"},
```
the field does *not* return the bip125 status, so at least the docs should say that (or maybe the field be removed completely)

Reply:
Removed the mention in the doc, i think we should still keep the deprecation period.
```diff
diff --git a/src/rpc/blockchain.cpp b/src/rpc/blockchain.cpp
index 337a10294b..18ab787f46 100644
--- a/src/rpc/blockchain.cpp
+++ b/src/rpc/blockchain.cpp
@@ -477,7 +477,7 @@ static std::vector<RPCResult> MempoolEntryDescription() { return {
         {RPCResult{RPCResult::Type::STR_HEX, "transactionid", "parent transaction id"}}},
     RPCResult{RPCResult::Type::ARR, "spentby", "unconfirmed transactions spending outputs from this transaction",
         {RPCResult{RPCResult::Type::STR_HEX, "transactionid", "child transaction id"}}},
-    RPCResult{RPCResult::Type::BOOL, "bip125-replaceable", "Whether this transaction could be replaced due to BIP125 (replace-by-fee) (DEPRECATED, use 'replaceable' field instead. Returned only if config option 'deprecatedrpc=bip125' is set)"},
+    RPCResult{RPCResult::Type::BOOL, "bip125-replaceable", "Whether this transaction could be replaced due to replace-by-fee (DEPRECATED, use 'replaceable' field instead. Returned only if config option 'deprecatedrpc=bip125' is set)"},
     RPCResult{RPCResult::Type::BOOL, "replaceable", "Whether this transaction could be replaced by a conflicting one using replace-by-fee"},
     RPCResult{RPCResult::Type::BOOL, "unbroadcast", "Whether this transaction is currently unbroadcast (initial broadcast not yet acknowledged by any peers)"},
 };}
diff --git a/src/wallet/rpcwallet.cpp b/src/wallet/rpcwallet.cpp
index 8339ff63d9..b66a77ea3d 100644
--- a/src/wallet/rpcwallet.cpp
+++ b/src/wallet/rpcwallet.cpp
@@ -1396,7 +1396,7 @@ static const std::vector<RPCResult> TransactionDescriptionString()
            {RPCResult::Type::NUM_TIME, "time", "The transaction time expressed in " + UNIX_EPOCH_TIME + "."},
            {RPCResult::Type::NUM_TIME, "timereceived", "The time received expressed in " + UNIX_EPOCH_TIME + "."},
            {RPCResult::Type::STR, "comment", "If a comment is associated with the transaction, only present if not empty."},
-           {RPCResult::Type::STR, "bip125-replaceable", "(\"yes|no\") Whether this transaction could be replaced due to BIP125 (replace-by-fee) (DEPRECATED, use 'replaceable' field instead. Returned only if config option 'deprecatedrpc=bip125' is set)"},
+           {RPCResult::Type::STR, "bip125-replaceable", "(\"yes|no\") Whether this transaction could be replaced due to replace-by-fee (DEPRECATED, use 'replaceable' field instead. Returned only if config option 'deprecatedrpc=bip125' is set)"},
            {RPCResult::Type::BOOL, "replaceable", "Whether this transaction could be replaced by a conflicting one using replace-by-fee"}};
 }
```

Reply:
Is it worth keeping `bip125-replaceable` returning the correct bip125 status for 1 release and then removing all the logic later?

Reply:
I think we should get rid of the misleading logic as soon as possible.